src/base/utils/Reaction.jl

```diff
@@ -39,34 +39,64 @@ function check_duplicate_reaction(
+Checks if on boundary by inspecting the number of metabolites in reaction equation.
 Boundary reactions have only one metabolite, e.g. an exchange reaction, or a sink/demand reaction.
 """
 function is_boundary(rxn::Reaction)::Bool
     length(keys(rxn.metabolites)) == 1 ? true : false
 end
 
+is_boundary(model::StandardModel, rxn_id::String) = is_boundary(model.reactions[rxn_id])
+
+is_boundary(model::StandardModel, rxn::Reaction) = is_boundary(rxn) # for consistency with functions below
+
 """
-    is_mass_balanced(model::StandardModel, rxn::Reaction)
+    reaction_atom_balance(model::StandardModel, rxn)
 
-Checks if `rxn` is atom balanced. Returns a boolean for whether the reaction is balanced,
-and the associated balance of atoms for convenience (useful if not balanced).
+Returns a dictionary mapping the stoichiometry of atoms through a single reaction. Uses the
+metabolite information in `model` to determine the mass balance. Accepts a reaction
+dictionary, a reaction string id or a `Reaction` as an argument for `rxn`.
 
-See also: [`get_atoms`](@ref), [`check_duplicate_reaction`](@ref)
+See also: [`reaction_is_mass_balanced`](@ref)
 """
-function is_mass_balanced(model::StandardModel, rxn::Reaction)
-    atom_balances = Dict{String,Float64}() # float here because stoichiometry is not Int
-    for (met, stoich) in rxn.metabolites
-        atoms = metabolite_formula(model, met)
-        isnothing(atoms) &&
+function reaction_atom_balance(model::StandardModel, reaction_dict::Dict{String,Float64})
+    atom_balances = Dict{String,Float64}()
+    for (met, stoich_rxn) in reaction_dict
+        adict = metabolite_formula(model, met)
+        isnothing(adict) &&
             throw(ErrorException("Metabolite $met does not have a formula assigned to it."))
-        for (k, v) in atoms
-            atom_balances[k] = get(atom_balances, k, 0) + v * stoich
+        for (atom, stoich_molecule) in adict
+            atom_balances[atom] =
+                get(atom_balances, atom, 0.0) + stoich_rxn * stoich_molecule
         end
     end
+    return atom_balances
+end
 
-    return all(sum(values(atom_balances)) == 0), atom_balances
+function reaction_atom_balance(model::StandardModel, rxn_id::String)
+    reaction_atom_balance(model, model.reactions[rxn_id].metabolites)
 end
+
+reaction_atom_balance(model::StandardModel, rxn::Reaction) =
+    reaction_atom_balance(model, rxn.id)
+
+"""
+    is_reaction_mass_balanced(model::StandardModel, rxn)
+
+Checks if `rxn` is atom balanced. Returns a boolean for whether the reaction is balanced,
+and the associated balance of atoms for convenience (useful if not balanced). Calls
+`reaction_atom_balance` internally.
+
+See also: [`check_duplicate_reaction`](@ref), [`reaction_atom_balance`](@ref)
+"""
+reaction_is_mass_balanced(model::StandardModel, rxn_id::String) =
+    all(values(reaction_atom_balance(model, rxn_id)) .== 0)
+
+reaction_is_mass_balanced(model::StandardModel, rxn::Reaction) =
```

Comment:
kinda lets me wonder if this will ever be needed, compare
```
reaction_mass_balanced(model, "asasd")
```
to
```
reaction_mass_balanced(model, model.reactions["asasd"])
```
...that is already 1 step from the 3rd overload with the reaction_dict aka stoichiometry, which is nicely generic.

In fact, are you sure this needs a `StandardModel` to work?

Reply:
yeah, I was wondering if it is a bit overkill to have as many overloads... but then again `reaction_mass_balanced` and `is_boundary` have all the overloads too, I was trying to be consistent... You decide, should I just keep the most basic one and let the user pass the reactions through `reaction_stoichiometry`?

Reply:
Let's keep it for now, is purely there for convenience